appserver/web/web-glue/src/main/java/com/sun/enterprise/web/PEAccessLogValve.java

```diff
@@ -1044,6 +1050,63 @@ private synchronized void open(String dateStamp,
         } 
 
     }
+    
+    /**
+     * Cleanup the history log file based on value set by <code>maxHistoryFiles</code>
+     * <p/>
+     * If it is not been specified without any value, then a default value of 10
+     * will be use.
+     * Else, if it is defined with value of 0, no history log file will be kept,
+     * and the current log file will be reset after each rotation.
+     * If undefined or value is less than 0, all history log are preserved. 
+     */
+    private void cleanUpHistoryLogFiles() {
+        if (maxHistoryFiles == 0 || maxHistoryFiles < 0) {
+            return;
+
+        }
+        
+        synchronized (lock) {
+            File fileDirectory = logFile.getParentFile();
+            String logFileName = logFile.getName();
+
+            if (fileDirectory == null) {
+                return;
+            }
+
+            File[] allFilesInDirectory = fileDirectory.listFiles();
+            ArrayList<String> candidateListOfLogFiles = new ArrayList<>();
+
+            if (allFilesInDirectory != null && allFilesInDirectory.length > 0) {
+                for (int i = 0; i < allFilesInDirectory.length; i++) {
+                    if (!logFileName.equals(allFilesInDirectory[i].getName())
+                            && allFilesInDirectory[i].isFile()
+                            && allFilesInDirectory[i].getName().startsWith(prefix)) {
+                        candidateListOfLogFiles.add(allFilesInDirectory[i].getAbsolutePath());
+                    }
+                }
+            }
+
+            if (candidateListOfLogFiles.size() <= maxHistoryFiles) {
+                return;
+            }
+
+            Object[] pathes = candidateListOfLogFiles.toArray();
+            Arrays.sort(pathes);
+            try {
+                for (int i = 0; i < pathes.length - maxHistoryFiles; i++) {
+                    File logFile = new File((String) pathes[i]);
+                    boolean delFile = logFile.delete();
+                    if (!delFile) {
+                        throw new IOException("Could not delete Access log file: "
```

Comment:
This seems harsh to do. Wouldn't it be better to write a log message here and continue removing log files. Otherwise a single undeletable log will stop log file removal forever.